src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskForSearchCompletionControllerTest.java

```diff
@@ -98,6 +117,125 @@ public void should_return_a_200_and_retrieve_a_task_by_event_and_case_match() {
         common.cleanUpTask(taskId, REASON_COMPLETED);
     }
 
+    @Test
+    public void should_return_a_200_and_retrieve_single_task_by_event_and_case_match_and_assignee() {
+        final String assigneeId = getAssigneeId(authenticationHeaders);
+
+        // create a caseId
+        final String caseId = given.iCreateACcdCase();
+
+        // create a 2 tasks for caseId
+        sendMessage(caseId);
+        sendMessage(caseId);
+
+        final List<CamundaTask> tasksList = iRetrieveATaskWithProcessVariableFilter("caseId", caseId, 2);
+        if (tasksList.size() != 2) {
+            fail("2 tasks should be created for case id: " + caseId);
+        }
+
+        // No user assigned to this task
+        final String taskId1 = tasksList.get(0).getId();
+
+        common.setupOrganisationalRoleAssignment(authenticationHeaders);
+        // assign user to taskId2
+        final String taskId2 = tasksList.get(1).getId();
+        // assign user to taskId2
```

Comment:
duplicate comment, do we need it?

Reply:
I am working on other related ticket, so will make changes.